app/src/main/java/com/randomappsinc/studentpicker/utils/SpeechUtil.java

```diff
@@ -0,0 +1,45 @@
+package com.randomappsinc.studentpicker.utils;
+
+import android.app.Activity;
+import android.content.ActivityNotFoundException;
+import android.content.Context;
+import android.content.Intent;
+import android.speech.RecognizerIntent;
+
+import com.randomappsinc.studentpicker.R;
+
+import java.util.List;
+import java.util.Locale;
+
+import static android.app.Activity.RESULT_OK;
+
+public class SpeechUtil {
+
+    public static final int SPEECH_REQUEST_CODE = 1;
+
+    public static void openSpeechToTextDialog(Activity activity) {
+        Intent intent = new Intent(RecognizerIntent.ACTION_RECOGNIZE_SPEECH);
+        intent.putExtra(RecognizerIntent.EXTRA_LANGUAGE_MODEL, RecognizerIntent.LANGUAGE_MODEL_FREE_FORM);
+        intent.putExtra(RecognizerIntent.EXTRA_LANGUAGE, Locale.getDefault());
+        intent.putExtra(RecognizerIntent.EXTRA_PROMPT, activity.getString(R.string.list_name_input_speech_message));
+        try {
+            activity.startActivityForResult(intent, SPEECH_REQUEST_CODE);
+            activity.overridePendingTransition(R.anim.slide_in_up, R.anim.slide_out_up);
+        } catch (ActivityNotFoundException exception) {
+            UIUtils.showLongToast(R.string.speech_not_supported, activity);
+        }
+    }
+
+    public static String processSpeechResults(int resultCode, Intent data, Context context) {
```

Comment:
We should mark this method as @Nullable so developers are aware that this can return null.
```suggestion
    @Nullable
    public static String processSpeechResults(int resultCode, Intent data, Context context) {
```